AwsEncryptionSDK/runtimes/go/examples/multithreading/awskmskeyring.go

```diff
@@ -0,0 +1,200 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+/*
+This example sets up the AWS KMS Keyring in an multithreaded environment.
+The AWS KMS keyring uses symmetric encryption KMS keys to generate, encrypt and
+decrypt data keys. This example creates a KMS Keyring and then encrypts a custom input exampleText
+with an encryption context. This example also includes some sanity checks for demonstration:
+1. Ciphertext and plaintext data are not the same
+2. Decrypted plaintext value matches exampleText
+These sanity checks are for demonstration in the example only. You do not need these in your code.
+AWS KMS keyrings can be used independently or in a multi-keyring with other keyrings
+of the same or a different type.
+For more information on how to use KMS keyrings, see
+https://docs.aws.amazon.com/encryption-sdk/latest/developer-guide/use-kms-keyring.html
+For more information on KMS Key identifiers, see
+https://docs.aws.amazon.com/kms/latest/developerguide/concepts.html#key-id
+*/
+
+package multithreading
+
+import (
+	"context"
+	"fmt"
+	"runtime"
+	"sync"
+
+	mpl "github.com/aws/aws-cryptographic-material-providers-library/mpl/awscryptographymaterialproviderssmithygenerated"
+	mpltypes "github.com/aws/aws-cryptographic-material-providers-library/mpl/awscryptographymaterialproviderssmithygeneratedtypes"
+	client "github.com/aws/aws-encryption-sdk/awscryptographyencryptionsdksmithygenerated"
+	esdktypes "github.com/aws/aws-encryption-sdk/awscryptographyencryptionsdksmithygeneratedtypes"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/kms"
+)
+
+// Structure to hold operation results
+type OperationResult struct {
+	EncryptOutput *esdktypes.EncryptOutput
+	DecryptOutput *esdktypes.DecryptOutput
+	Error         error
+}
```

Comment:
Are you exposing this to outside of the package?

Reply:
Removed OperationResult and returned `result, error` in the function call